org.eclipse.m2e.discovery/META-INF/MANIFEST.MF

```diff
@@ -2,7 +2,7 @@ Manifest-Version: 1.0
 Bundle-ManifestVersion: 2
 Bundle-Name: %Bundle-Name
 Bundle-SymbolicName: org.eclipse.m2e.discovery;singleton:=true
-Bundle-Version: 1.18.0.qualifier
+Bundle-Version: 1.18.2.qualifier
```

Comment:
Agreed. Regarind Version Bumps: Is there an integrated/automated version bump, because I cannot remember ever having set the version to 1.18.2 myself.
Additionally: What do you think should the bumped m2e.core version be. Strictly speaking it should be 2.0.0, right?

Reply:
> Agreed. Regarind Version Bumps: Is there an integrated/automated version bump, because I cannot remember ever having set the version to 1.18.2 myself.

Actually there is nothing installed through m2e's Oomph setup that can bump versions automatically and I don't know any plug-in/tool that bumps automatically. There are the PDE API-Tools that can check for required bumps, but they don't bump automatically and m2e is not configured for them.

> Additionally: What do you think should the bumped m2e.core version be. Strictly speaking it should be 2.0.0, right?

No need for that. All classes you have changed are internal (note the `internal` segment in the package name) and not part of the public API and therefore can be changed arbitrarily without further notice. So a micro/service version bump by one is sufficient.

Reply:
The only thing that need to be coordinated are the tests in `m2e-core-tests`. These are actually in another repository and included as git submodule. So the changes to them have to be addressed to another repository (https://github.com/tesla/m2e-core-tests). This structure was grown historically.

But we have a chicken-egg-problem here, since the test fail if you change the method-signature here but not in the tests and you cannot change the tests first because they would fail with compile errors because they are compiled against the current methods.
So either: 
- you first only add corresponding method overloads with additional context argument and keep the old methods delegating to the new ones and passing null as context (then perform a null check before using the context, but you should to this anyway), then update the tests to use the new methods with another PR to the mentioned repo and once the second PR is merged, create a third PR to remove the old methods.
- or parallel to this PR the PR for the tests repo is created and once both are ready to be merged (and we are sure everything will work fine) we merge them immediately one after the other and ignore that builds fail for a short time.

@mickaelistria do you have a preferred way how this is handled?